core/lib/PathFunctor.agda

```diff
@@ -302,46 +311,51 @@ module _ {i} {A : Type i} where
     ap (λ w → w ∙ h y) (! (ap-idf p)) ∙ =ₛ-out (homotopy-naturality (λ a → a) g h p)
 
   homotopy-naturality-to-cst : ∀ {k} {B : Type k} (f : A → B) (b : B)
+    (h : (x : A) → f x == b) {x y : A} (p : x == y)
+    → ap f p ◃∙ h y ◃∎ =ₛ h x ◃∎
+  homotopy-naturality-to-cst f b h p@idp = =ₛ-in idp
+
+  homotopy-to-cst-ap : ∀ {k} {B : Type k} (f : A → B) (b : B)
     (h : (x : A) → f x == b) {x y : A} (p : x == y)
     → ap f p == h x ∙ ! (h y)
-  homotopy-naturality-to-cst f b h {x} p@idp = ! (!-inv-r (h x))
+  homotopy-to-cst-ap f b h {x} p@idp = ! (!-inv-r (h x))
```

Comment:
What does the ending `ap` mean? It seems `naturality` also involves `ap`. I am not against this, but I would like to learn the rationale.

Reply:
I chose the suffix `-ap` because the statement involves `ap`. I'm not happy with this choice either. Two suggestions:

* `homotopy-naturality-to-cst'` would make sense since the statement is similar to ``homotopy-naturality-to-cst` (with the paths "rotated").
* `ap-homotopic-to-cst` would make sense, since the definition can be seen as the generalization of `ap-cst` where the function is not judgmentally the constant function, but only homotopic to it. (This is my favorite.)

Reply:
Sorry for the late reply. I am more worried about consistency, and I think both are compatible with the documented convention. (In fact, I think the intention was to adopt cubical methods instead of naturality lemmas.) I wonder how one can generalize the `homotopic` so that it can be registered as a general principle.

Reply:
Another proposal: `ap-null-homotopic`

Citing [Wikipedia](https://en.wikipedia.org/wiki/Homotopy#Null-homotopy):

> A function f is said to be null-homotopic if it is homotopic to a constant function.

I am not sure what kind of generalization you are seeking. In our naming convention, `null-homotopic` (or `homotopic-to-cst`) could be used everywhere where `cst` or `idf` is used.

Reply:
I was worried about the parsing of a name involving `to`. In general, `A-to-B` is the (canonical) function from `A` to `B`. In this case, it is fine because `cst` is not a type, but I do not know what to say if we are going to consider `homotopic-to-X` for generic `X`. `null-homotopic` also works because there is no confusion in special cases.

Reply:
I've renamed the function to `ap-null-homotopic`.